pkgs/by-name/pi/pihole-ftl/package.nix

```diff
@@ -0,0 +1,66 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, gmp
+, libidn
+, nettle
+, readline
+, ncurses
+, xxd
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "pihole-ftl";
+  version = "5.23";
+
+  src = fetchFromGitHub {
+    owner = "pi-hole";
+    repo = "FTL";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-kMOGf6Qsayhr7uuFp9cGqJFubpFwlhJjVxEdpC1mO/M=";
+  };
+
+  patches = [
+    ./0000-fix-readline-linking.patch
+    # Nix enables fortification, avoid duplicating the compile definition
+    ./0001-remove-fortify-source.patch
+  ];
+
+  nativeBuildInputs = [ cmake xxd ];
+
+  buildInputs = [ gmp libidn nettle readline ncurses ];
+
+  postPatch = ''
+    # Swap the static library suffix options to shared for non-static builds
+    ${lib.optionalString (!stdenv.hostPlatform.isStatic) ''
+    substituteInPlace src/CMakeLists.txt \
+      --replace "CMAKE_STATIC_LIBRARY_SUFFIX" "CMAKE_SHARED_LIBRARY_SUFFIX"
+    ''}
```

Comment:
```suggestion
  postPatch =
    # Swap the static library suffix options to shared for non-static builds
    lib.optionalString (!stdenv.hostPlatform.isStatic) ''
      substituteInPlace src/CMakeLists.txt \
        --replace "CMAKE_STATIC_LIBRARY_SUFFIX" "CMAKE_SHARED_LIBRARY_SUFFIX"
    '' ++ ''
```

Reply:
Went for a list, Nix complained about concatenating strings like this.